### PR TITLE
container inspect show labels

### PR DESCRIPTION
**Pull request by TianZong48**
Signed-off-by: tianyang ni <tianzong48@gmail.com>

https://github.com/containerd/nerdctl/issues/1261

```
# nerdctl run -d --name foo --label foo=bar alpine sleep infinity
# nerdctl container inspect foo |jq ".[0].Config.Labels"
{
  "foo": "bar"
}
```

### CODE REVIEW

File: pkg/inspecttypes/dockercompat/dockercompat.go [hunk 1]
```diff
@@ -273,9 +273,37 @@ func ContainerFromNative(n *native.Container) (*Container, error) {
 		}
 		c.NetworkSettings = nSettings
 	}
+
+	c.Config = &Config{
+		Hostname: n.Labels[labels.Hostname],
+		Labels: filterLabels(n.Labels,
+			hasPrefix(labels.Prefix),
```
Comment (TianZong48) [inline pkg/inspecttypes/dockercompat/dockercompat.go:280]: trying to hide some labels used by nerdctl or added by containerd

https://github.com/containerd/nerdctl/pull/1260/files#r929190294

File: pkg/inspecttypes/dockercompat/dockercompat.go [hunk 2]
```diff
@@ -273,9 +273,38 @@ func ContainerFromNative(n *native.Container) (*Container, error) {
...
 	}
+
+	c.Config = &Config{
+		Hostname: n.Labels[labels.Hostname],
+		Labels: filterLabels(n.Labels,
+			// filter out internal labels
+			hasPrefix(labels.Prefix),
+			hasPrefix("containerd.io"), // https://github.com/containerd/containerd/blob/release/1.6/runtime/restart/restart.go#L42-L50
+			hasPrefix(containerd.StopSignalLabel)),
+	}
```
Comment (AkihiroSuda) [inline pkg/inspecttypes/dockercompat/dockercompat.go:284]: I don't think we should filter out internal labels.
Happy to discuss in a separate issue/PR though.

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
https://github.com/containerd/nerdctl/pull/1265/files#r930080531

**Comment by AkihiroSuda** [comment]
Please squash commits

File: cmd/nerdctl/container_inspect_linux_test.go [hunk 3]
```diff
@@ -138,3 +138,17 @@ func TestContainerInspectContainsMounts(t *testing.T) {
 		}
 	}
 }
+
+func TestContainerInspectContainsLabel(t *testing.T) {
```
Comment (junnplus) [inline cmd/nerdctl/container_inspect_linux_test.go:142]: Would we can parallel test?

Comment (TianZong48) [reply cmd/nerdctl/container_inspect_linux_test.go:142]: oh yeah, I was thining about it and forgot haha.

File: cmd/nerdctl/container_inspect_linux_test.go [hunk 4]
```diff
@@ -138,3 +138,19 @@ func TestContainerInspectContainsMounts(t *testing.T) {
...
+	t.Parallel()
+	testContainer := testutil.Identifier(t)
+
+	base := testutil.NewBase(t)
+	defer base.Cmd("rm", "-f", testContainer).Run()
+
+	base.Cmd("run", "-d", "--name", testContainer, "--label", "foo=foo", "--label", "bar=bar", testutil.NginxAlpineImage).AssertOK()
+	base.EnsureContainerStarted(testContainer)
```
Comment (junnplus) [inline cmd/nerdctl/container_inspect_linux_test.go:150]: nit: not required

**Review by junnplus — APPROVED** [review]
Thanks, LGTM on green

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**